### PR TITLE
feat(toast): quickly swiping dismisses toast

### PR DESCRIPTION
**Pull request by liamdebeasi**
Issue number: N/A

---------

## What is the current behavior?

The swipeable toast dismiss animation does not change based on how quickly the toast was swiped. This makes the animation seem sluggish.

## What is the new behavior?

- The toast now dismisses quicker depending on how quickly users swipe.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: core/src/components/toast/gestures/swipe-to-dismiss.ts [hunk 1]
```diff
@@ -186,46 +184,65 @@ export const createSwipeToDismissGesture = (
...
       const KEYFRAMES = [
         { offset: 0, transform: `translateY(${startPosition})` },
         { offset: 1, transform: `translateY(${endPosition})` },
       ];
 
       swipeAnimation.keyframes(KEYFRAMES);
+
+      /**
+       * Compute the remaining amount of pixels the
+       * to toast needs to move to be fully dismissed.
```
Comment (thetaPC) [inline core/src/components/toast/gestures/swipe-to-dismiss.ts:221]:
```suggestion
       * toast needs to move to be fully dismissed.
```

File: core/src/components/toast/gestures/swipe-to-dismiss.ts [hunk 2]
```diff
@@ -186,46 +184,65 @@ export const createSwipeToDismissGesture = (
...
+       */
+      remainingDistance = endOffset - startOffset;
     } else {
       shouldDismiss = threshold >= DISMISS_THRESHOLD;
       playTo = shouldDismiss ? 1 : 0;
       step = computeStep(detail.deltaY);
+
+      /**
+       * Compute the remaining amount of pixels the
+       * to toast needs to move to be fully dismissed.
```
Comment (thetaPC) [inline core/src/components/toast/gestures/swipe-to-dismiss.ts:231]:
```suggestion
       * toast needs to move to be fully dismissed.
```

**Review by thetaPC — APPROVED** [review]
LGTM, works great! Matches my speed well.

**Review by sean-perkins — APPROVED** [review]
This feels really slick on device. Nice work!

**Pull request merged**